qiskit/qasm2/__init__.py

```diff
@@ -108,6 +125,59 @@
 Examples
 ========
 
+Exporting examples
+------------------
+
+Export a simple :class:`.QuantumCircuit` to an OpenQASM 2 string:
+
+.. code-block:: python
+
+    import qiskit.qasm2
+    from qiskit.circuit import QuantumCircuit
+
+    qc = QuantumCircuit(2, 2)
+    qc.h(0)
+    qc.cx(0, 1)
+    qc.measure([0, 1], [0, 1])
+    print(qiskit.qasm2.dumps(qc))
+
+.. code-block:: text
+
+    OPENQASM 2.0;
+    include "qelib1.inc";
+    qreg q[2];
+    creg c[2];
+    h q[0];
+    cx q[0],q[1];
+    measure q[0] -> c[0];
+    measure q[1] -> c[1];
+
+Write out the same :class:`.QuantumCircuit` to a given filename:
+
+.. code-block:: python
+
+    qiskit.qasm2.dump(qc, "myfile.qasm")
+
+Similarly, one can use general :class:`os.PathLike` instances as the filename:
+
+.. code-block:: python
+
+    import pathlib
+
+    qiskit.qasm2.dump(qc, pathlib.Path.home() / "myfile.qasm")
```

Comment:
I didnt know that `pathlib.Path.home() / "myfile.qasm"` was possible. Nice one!

Reply:
Yeah, `pathlib` is typically a much more convenient interface to working with filesystems than raw `os.path`, and almost everything that deals with path-like objects will accept `T | os.PathLike[T]` as the type (for `T: str | bytes`), of which `pathlib.Path` is one.